test/parallel/test-runner-run.mjs

```diff
@@ -551,3 +551,9 @@ describe('forceExit', () => {
     });
   });
 });
+
+// exitHandler doesn't run until after the tests / after hooks finish.
+process.on('exit', () => {
```

Comment:
What about `'beforeExit'`, `'SIGINT'`, and `'SIGTERM'`?

Reply:
Good point, I added checks for those as well.